mqtt/mqtt-broker/src/server.rs

```diff
@@ -163,14 +168,16 @@ where
     }
 }
 
-async fn incoming_task<A, F>(
+async fn incoming_task<A, F, N>(
     transport: TransportBuilder<A>,
     handle: BrokerHandle,
     mut shutdown_signal: F,
+    authenticator: N,
```

Comment:
nit: you can pass a ref `&N` to avoid extra clone.

Reply:
this is how it works with the clones:
- at line 56 it creates its own clone for every transport , which the will be passed to their respective incoming_task (e.g. if there is 2 transports specified, now both incoming_task will have its own copy.
- then the given incoming_task has a loop around incoming connections, and every new connection has its own copy of authorizator.

I don't see how reference could work?

Reply:
we can probably wrap it with Arc and clone references?
```suggestion
    authenticator: Arc<N>,
```